### PR TITLE
fix(index): handle `module` extraction correctly (`require.ensure`)

### PR DESCRIPTION
**Pull request by lvming6816077**
### `Issues`

- Fixes https://github.com/webpack-contrib/extract-text-webpack-plugin/issues/686

### CODE REVIEW

**Comment by lvming6816077** [comment]
TENNYLV-MB0:extract-text-webpack-plugin lvming$ npm test

> extract-text-webpack-plugin@3.0.2 test /Users/lvming/Desktop/work/extract-text-webpack-plugin
> jest

 PASS  test/define-fallback.test.js
 PASS  test/define-loader.test.js
 PASS  test/schema-validation.test.js
 PASS  test/extract-text-plugin.test.js

 RUNS  test/webpack-integration.test.js
 PASS  test/webpack-integration.test.js

Test Suites: 5 passed, 5 total
Tests:       39 passed, 39 total
Snapshots:   26 passed, 26 total
Time:        5.042s
Ran all test suites.

**Comment by michael-ciniawsky** [comment]
@evilebottnawi There was a switch made to CircleCI for `next` and it seems `master` isn't enabled to for `travis` anymore, which needs to be fixed somehow...

**Comment by evilebottnawi** [comment]
@michael-ciniawsky locally test passed, seems we can approved and shipped

File: src/index.js [hunk 1]
```diff
@@ -143,7 +143,7 @@ class ExtractTextPlugin {
         });
         async.forEach(chunks, (chunk, callback) => { // eslint-disable-line no-shadow
           const extractedChunk = extractedChunks[chunks.indexOf(chunk)];
-          const shouldExtract = !!(options.allChunks || isInitialOrHasNoParents(chunk));
+          const shouldExtract = !!(options.allChunks);
```
Comment (michael-ciniawsky) [inline src/index.js]: Unnecessary parentheses

File: src/index.js [hunk 2]
```diff
@@ -152,8 +152,17 @@ class ExtractTextPlugin {
               // A stricter `shouldExtract !== wasExtracted` check to guard against cases where a previously extracted
               // module would be extracted twice. Happens when a module is a dependency of an initial and a non-initial
               // chunk. See issue #604
-              if (shouldExtract && !wasExtracted) {
-                module[`${NS}/extract`] = shouldExtract; // eslint-disable-line no-path-concat
+
+              const cks = module.chunks;
```
Comment (michael-ciniawsky) [inline src/index.js]: `cks` => `chunks` or just use `module.chunks` directly

File: src/index.js [hunk 3]
```diff
@@ -152,8 +152,17 @@ class ExtractTextPlugin {
...
+              // check every module's chunks.parents() to decide extract or not
+              for (let i = 0; i < cks.length; i++) {
+                if (!isInitialOrHasNoParents(cks[i]) && !module.noExtracted) {
+                  module.noExtracted = 1;
```
Comment (michael-ciniawsky) [inline src/index.js]: `module.noExtracted`  => `module.extracted` and use a `{Boolean}`  e.g

```js
if (...) {
  module.extracted = true;

  break;
}
```

**Comment by lvming6816077** [comment]
@michael-ciniawsky thanks for your suggestion，I have committed the code refine，please check！

**Comment by evilebottnawi** [comment]
@michael-ciniawsky ready for merge :+1:

**Comment by michael-ciniawsky** [comment]
@lvming6816077 I will try to get this out as soon as possible, but I need to do some chores beforehand and merge other outstanding PR's aswell (ETA ~1-2 days)

**Pull request merged**